client/js/components/registration-form.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import TextField from 'material-ui/lib/text-field';
+import Dialog from 'material-ui/lib/dialog';
+import RaisedButton from 'material-ui/lib/raised-button';
+import UserActions from '../actions/user-actions';
+
+export default class RegistrationForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this._onRequestClose = this._onRequestClose.bind(this);
+    this._onSignUpRequest = this._onSignUpRequest.bind(this);
+  }
+
+  _onRequestClose() {
+    this.refs.dialog.setState({open: true});
```

Comment:
@Vakrim I'm not sure that I understand but you forcing that element to be open event the component was requested to be closed. But I can't see possibility to close request close of that ```Dialog``` component (is it on click on overlay? if so there is option to disable it.)